### ISSUE TITLE
possible to train without mask?

### ISSUE THREAD

**Issue by cchamber** [anatolix/keras_Realtime_Multi-Person_Pose_Estimation#21]
I have a new dataset I would like to train on but do not have segmentation masks. Is it possible to train without the masks?

**Comment by abinjoabraham** [comment]
Same question I am also having. Can we train without the masks. Actually what is the main need for the masks ? Can we implement the 2D pose by just the key points?

**Comment by cchamber** [comment]
I trained using a bounding box instead of a mask and gave the coordinates of the box instead of the segmentation polygon. I only used images without occlusion for training. This seems to work.

**Comment by abinjoabraham** [comment]
@cchamber thanks buddy. Can I look on your code ? Is the fork available online. ?

**Comment by murrayLuke** [comment]
@abinjoabraham the masks seem to be used to filter out unlabeled people from the dataset. For example if there is a crowd of people maybe three of them will have body parts labeled. The mask then covers the remaining people in the crowd. When the algorithm recognizes a body part on someone who is masked out, it is ignored, since there is no ground truth.

**Comment by typhon3d** [comment]
Thus, when the dataset includes only images with one single person, the mask is redundant? just a full white image can be used for the mask and the results of the model will be of equal accuracy?

**Comment by cchamber** [comment]
@abinjoabraham I made a couple of edits to Anatolix's code https://github.com/cchamber/openpose_keras
There's also an annotation tool for video. you can convert your labels made with that to COCO for training
https://github.com/cchamber/convert_videolabels_to_NNinput
if you want to build a mask you can take the area included within the keypoints (convex hull) with a bit of padding around it

**Comment by abinjoabraham** [comment]
@cchamber Thank you for sharing your version. I will surely study on it. Being a newbie to python and deep learning world these codes are giving me very good knowledge in Artificial Intelligence.

**Comment by 1209805200** [comment]
@cchamber the mask is necessary? i am not sure the groundtrue include  what?keypoint and mask??? hope your reply
